float64[] Robot_limits

JointLimit[] JointLimitdata 

SafetyPlane[] SafetyPlanedata

Tool[]   Tooldata

int32[] SafetyInputIo

bool underSingularity
bool too_close_point 
bool safety_flag


